tests/bin/agent/lock.test.ts

```diff
@@ -9,7 +9,7 @@ import * as testBinUtils from '../utils';
 import * as testUtils from '../../utils';
 
 jest.mock('prompts');
-const mockedPrompts = mocked(prompts);
```

Comment:
I don't understand this?

We are mocking the `prompts` module entirely. See `jest.mock('prompt's);` at the beginning.

Why is this now `mocked(prompts.prompt)`? That would imply it's mocking something else entirely. It's mocking the `prompt` property of `prompts` module. But we want to mock the `prompts` module itself.

Reply:
https://github.com/MatrixAI/js-polykey/pull/394#issuecomment-1170762203

Reply:
I mean I can see the type error that comes from using `mockImplementation`, but perhaps this is not meant to be solved by using `prompts.prompt`, but instead a different `mock*` method?

Reply:
I tried using other mock methods but they had the same error, or otherwise caused test failures. The part of the `prompts` library that we're mocking is just the `prompt` method, so I think mocking `prompts.prompt` is correct

Reply:
I say this because the TypeError is saying that the implementation we're providing is missing properties of `prompts`, `prompt` being one of them, so if we were to mock all of `prompts` we would need to provide an implementation for all of the other methods as well (e.g. `inject` and `override`). We only need to use `prompts.prompt` so we only need to mock that part.

Reply:
Oh ok... still seems like then `jest.mock('prompts')` seems incorrect then. But if it works right now, then we can proceed...

Did you test the relevant tests too to see if `mockedPrompts` ended up working?

Reply:
Yeah, after making the change our mocks work as expected